### ISSUE TITLE
App crash on MapView render when running alongside Detox

### ISSUE THREAD

**Issue by bgneu** [wix/Detox#2516]
### Description
I have a react-native project which uses `react-native-maps` and is setup to display Google Maps rather than the default native map, in this case Apple Maps. My team and I have just made the decision to integrate detox for e2e testing, however the app crashes whenever a test moves on to a screen that renders a MapView.

From what I could tell, it seems that the Google Maps framework crashes on attempting to load tiles and POIs when running alongside Detox. 

Other things I have tried: 
1. Setup `react-native-maps` using the native maps setup, in this case Apple Maps - Crash **DID NOT** occur.
2. Run app using an invalid API Key for Google Maps - Map does not contain any tiles or information other than the Google logo, however the app **DID NOT** crash.
3. [Manual Debugging in Xcode](https://github.com/wix/Detox/blob/master/docs/Guide.DebuggingInXcode.md) - I was able to repro the crash and breakpoint on the exception, but Xcode did not provide anything useful/actionable. 

- [x] I have tested this issue on the latest Detox release and it still reproduces

#### Reproduction

I have created a repo that uses the bare minimum setup for both `react-native-maps` and `detox` to repro the issue with as little bloat as possible.

To reproduce:
1. Check out the repo [here](https://github.com/bgneu/DetoxTest).
2. On project root, run `yarn install`.
3. On project root, run `cd ios && pod install && cd ..`.
4. In the AppDelegate.m, replace `<YOUR_API_KEY>` (line 35) with your valid API Key provided by Google. 
5. On project root, run `detox build -c ios`.
6. On project root, run `detox test -c ios`.
7. Project should crash immediately.

#### Expected behavior
Simply put, the MapView should render correctly and tests should pass without a problem.

#### Screenshots
N/A

#### Environment (please complete the following information):
 - Detox: 17.14.3
 - React Native: 0.63.4
 - Node: 14.4.0
 - Device: iPhone Simulator
 - Xcode: 12.2
 - iOS: 14.2
 - macOS: Catalina (10.15.7)

#### If you are experiencing a timeout in your test
- [ ] I have followed the instructions under [Lower-level Idling Resources Debug](https://github.com/wix/Detox/blob/master/docs/Troubleshooting.Synchronization.md#lower-level-idling-resources-debug-ios-only) and am providing the log below: N/A

#### If you are seeing a build problem (e.g. during `npm install`)
- [ ] I am providing the `npm install` log below: N/A

#### Device and verbose Detox logs
- [x] I have run my tests using the `--loglevel trace` argument and am providing the verbose log below:
```
developer$ detox test -c ios -l trace
detox[82519] INFO:  [test.js] configuration="ios" loglevel="trace" useCustomLogger=true DETOX_START_TIMESTAMP=1607454804660 reportSpecs=true jest --config e2e/config.json --testNamePattern '^((?!:android:).)*$' --maxWorkers 1 e2e
detox[82520] TRACE: [Detox.js/DETOX_CREATE] created a Detox instance with config:
{"artifactsConfig":{"rootDir":"artifacts/ios.2020-12-08 19-13-24Z","plugins":{"log":{"enabled":false,"keepOnlyFailedTestsArtifacts":false},"screenshot":{"enabled":true,"shouldTakeAutomaticSnapshots":false,"keepOnlyFailedTestsArtifacts":false},"video":{"enabled":false,"keepOnlyFailedTestsArtifacts":false},"instruments":{"enabled":false,"keepOnlyFailedTestsArtifacts":false},"timeline":{"enabled":false}},"pathBuilder":{"_rootDir":"artifacts/ios.2020-12-08 19-13-24Z"}},"behaviorConfig":{"init":{"reinstallApp":true,"exposeGlobals":true,"launchApp":true},"cleanup":{"shutdownDevice":false}},"cliConfig":{"configuration":"ios","loglevel":"trace","useCustomLogger":"true"},"deviceConfig":{"type":"ios.simulator","binaryPath":"ios/build/Build/Products/Debug-iphonesimulator/TestProject.app","build":"xcodebuild -workspace ios/TestProject.xcworkspace -scheme TestProject -sdk iphonesimulator -derivedDataPath ios/build","device":{"type":"iPhone 11"}},"runnerConfig":{"testRunner":"jest","runnerConfig":"e2e/config.json","specs":"e2e"},"sessionConfig":{"autoStart":true,"server":"ws://localhost:56094","sessionId":"3df58e7d-a33e-48b9-fa67-ec0177e11763","debugSynchronization":false},"errorBuilder":{"filepath":"/Users/developer/Workspace/DetoxTest/.detoxrc.json","contents":{"testRunner":"jest","runnerConfig":"e2e/config.json","configurations":{"ios":{"type":"ios.simulator","binaryPath":"ios/build/Build/Products/Debug-iphonesimulator/TestProject.app","build":"xcodebuild -workspace ios/TestProject.xcworkspace -scheme TestProject -sdk iphonesimulator -derivedDataPath ios/build","device":{"type":"iPhone 11"}},"android":{"type":"android.emulator","binaryPath":"SPECIFY_PATH_TO_YOUR_APP_BINARY","device":{"avdName":"Pixel_2_API_29"}}}},"configurationName":"ios"}}
detox[82520] INFO:  [DetoxServer.js] server listening on localhost:56094...
detox[82520] DEBUG: [AsyncWebSocket.js/WEBSOCKET_OPEN] opened web socket to: ws://localhost:56094
detox[82520] TRACE: [AsyncWebSocket.js/WEBSOCKET_SEND] {"type":"login","params":{"sessionId":"3df58e7d-a33e-48b9-fa67-ec0177e11763","role":"tester"},"messageId":0}
detox[82520] DEBUG: [DetoxServer.js/LOGIN] role=tester, sessionId=3df58e7d-a33e-48b9-fa67-ec0177e11763
detox[82520] DEBUG: [DetoxServer.js/LOGIN_SUCCESS] role=tester, sessionId=3df58e7d-a33e-48b9-fa67-ec0177e11763
detox[82520] TRACE: [AsyncWebSocket.js/WEBSOCKET_MESSAGE] {"type":"loginSuccess","params":{"sessionId":"3df58e7d-a33e-48b9-fa67-ec0177e11763","role":"tester"},"messageId":0}

detox[82520] DEBUG: [exec.js/EXEC_CMD, #0] applesimutils --list --byType "iPhone 11"
detox[82520] TRACE: [exec.js/EXEC_SUCCESS, #0] [
  {
    "os" : {
      "bundlePath" : "\/Applications\/Xcode.app\/Contents\/Developer\/Platforms\/iPhoneOS.platform\/Library\/Developer\/CoreSimulator\/Profiles\/Runtimes\/iOS.simruntime",
      "buildversion" : "18B79",
      "runtimeRoot" : "\/Applications\/Xcode.app\/Contents\/Developer\/Platforms\/iPhoneOS.platform\/Library\/Developer\/CoreSimulator\/Profiles\/Runtimes\/iOS.simruntime\/Contents\/Resources\/RuntimeRoot",
      "identifier" : "com.apple.CoreSimulator.SimRuntime.iOS-14-2",
      "version" : "14.2",
      "isAvailable" : true,
      "name" : "iOS 14.2"
    },
    "dataPath" : "\/Users\/developer\/Library\/Developer\/CoreSimulator\/Devices\/FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C\/data",
    "logPath" : "\/Users\/developer\/Library\/Logs\/CoreSimulator\/FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C",
    "udid" : "FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C",
    "isAvailable" : true,
    "deviceType" : {
      "minRuntimeVersion" : 851968,
      "bundlePath" : "\/Applications\/Xcode.app\/Contents\/Developer\/Platforms\/iPhoneOS.platform\/Library\/Developer\/CoreSimulator\/Profiles\/DeviceTypes\/iPhone 11.simdevicetype",
      "maxRuntimeVersion" : 4294967295,
      "name" : "iPhone 11",
      "identifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-11",
      "productFamily" : "iPhone"
    },
    "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-11",
    "state" : "Booted",
    "name" : "iPhone 11"
  },
  {
    "os" : {
      "bundlePath" : "\/Library\/Developer\/CoreSimulator\/Profiles\/Runtimes\/iOS 13.7.simruntime",
      "buildversion" : "17H22",
      "runtimeRoot" : "\/Library\/Developer\/CoreSimulator\/Profiles\/Runtimes\/iOS 13.7.simruntime\/Contents\/Resources\/RuntimeRoot",
      "identifier" : "com.apple.CoreSimulator.SimRuntime.iOS-13-7",
      "version" : "13.7",
      "isAvailable" : true,
      "name" : "iOS 13.7"
    },
    "dataPath" : "\/Users\/developer\/Library\/Developer\/CoreSimulator\/Devices\/EDBA6C98-B33B-4DC6-AF17-FB5F9E2DF683\/data",
    "logPath" : "\/Users\/developer\/Library\/Logs\/CoreSimulator\/EDBA6C98-B33B-4DC6-AF17-FB5F9E2DF683",
    "udid" : "EDBA6C98-B33B-4DC6-AF17-FB5F9E2DF683",
    "isAvailable" : true,
    "deviceType" : {
      "minRuntimeVersion" : 851968,
      "bundlePath" : "\/Applications\/Xcode.app\/Contents\/Developer\/Platforms\/iPhoneOS.platform\/Library\/Developer\/CoreSimulator\/Profiles\/DeviceTypes\/iPhone 11.simdevicetype",
      "maxRuntimeVersion" : 4294967295,
      "name" : "iPhone 11",
      "identifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-11",
      "productFamily" : "iPhone"
    },
    "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-11",
    "state" : "Shutdown",
    "name" : "iPhone 11"
  }
]

detox[82520] DEBUG: [exec.js/EXEC_CMD, #1] applesimutils --list --byId FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C --maxResults 1
detox[82520] TRACE: [exec.js/EXEC_SUCCESS, #1] [
  {
    "os" : {
      "bundlePath" : "\/Applications\/Xcode.app\/Contents\/Developer\/Platforms\/iPhoneOS.platform\/Library\/Developer\/CoreSimulator\/Profiles\/Runtimes\/iOS.simruntime",
      "buildversion" : "18B79",
      "runtimeRoot" : "\/Applications\/Xcode.app\/Contents\/Developer\/Platforms\/iPhoneOS.platform\/Library\/Developer\/CoreSimulator\/Profiles\/Runtimes\/iOS.simruntime\/Contents\/Resources\/RuntimeRoot",
      "identifier" : "com.apple.CoreSimulator.SimRuntime.iOS-14-2",
      "version" : "14.2",
      "isAvailable" : true,
      "name" : "iOS 14.2"
    },
    "dataPath" : "\/Users\/developer\/Library\/Developer\/CoreSimulator\/Devices\/FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C\/data",
    "logPath" : "\/Users\/developer\/Library\/Logs\/CoreSimulator\/FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C",
    "udid" : "FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C",
    "isAvailable" : true,
    "deviceType" : {
      "minRuntimeVersion" : 851968,
      "bundlePath" : "\/Applications\/Xcode.app\/Contents\/Developer\/Platforms\/iPhoneOS.platform\/Library\/Developer\/CoreSimulator\/Profiles\/DeviceTypes\/iPhone 11.simdevicetype",
      "maxRuntimeVersion" : 4294967295,
      "name" : "iPhone 11",
      "identifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-11",
      "productFamily" : "iPhone"
    },
    "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-11",
    "state" : "Booted",
    "name" : "iPhone 11"
  }
]

detox[82520] TRACE: [ArtifactsManager.js/LIFECYCLE] artifactsManager.onBootDevice({
  coldBoot: false,
  deviceId: 'FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C',
  type: 'iPhone 11'
})
detox[82520] TRACE: [ArtifactsManager.js/LIFECYCLE] artifactsManager.onBeforeUninstallApp({
  deviceId: 'FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C',
  bundleId: 'com.testproject.detox'
})
detox[82520] DEBUG: [exec.js/EXEC_CMD, #2] /usr/bin/xcrun simctl uninstall FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C com.testproject.detox
detox[82520] DEBUG: [exec.js/EXEC_TRY, #2] Uninstalling com.testproject.detox...
detox[82520] TRACE: [exec.js/EXEC_SUCCESS, #2]
detox[82520] DEBUG: [exec.js/EXEC_SUCCESS, #2] com.testproject.detox uninstalled
detox[82520] DEBUG: [exec.js/EXEC_CMD, #3] /usr/bin/xcrun simctl install FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C "/Users/developer/Workspace/DetoxTest/ios/build/Build/Products/Debug-iphonesimulator/TestProject.app"
detox[82520] DEBUG: [exec.js/EXEC_TRY, #3] Installing /Users/developer/Workspace/DetoxTest/ios/build/Build/Products/Debug-iphonesimulator/TestProject.app...
detox[82520] TRACE: [exec.js/EXEC_SUCCESS, #3]
detox[82520] DEBUG: [exec.js/EXEC_SUCCESS, #3] /Users/developer/Workspace/DetoxTest/ios/build/Build/Products/Debug-iphonesimulator/TestProject.app installed
detox[82520] TRACE: [ArtifactsManager.js/LIFECYCLE] artifactsManager.onBeforeTerminateApp({
  deviceId: 'FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C',
  bundleId: 'com.testproject.detox'
})
detox[82520] DEBUG: [exec.js/EXEC_CMD, #4] /usr/bin/xcrun simctl terminate FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C com.testproject.detox
detox[82520] DEBUG: [exec.js/EXEC_TRY, #4] Terminating com.testproject.detox...
detox[82520] TRACE: [exec.js/EXEC_SUCCESS, #4]
detox[82520] DEBUG: [exec.js/EXEC_SUCCESS, #4] com.testproject.detox terminated
detox[82520] TRACE: [ArtifactsManager.js/LIFECYCLE] artifactsManager.onTerminateApp({
  deviceId: 'FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C',
  bundleId: 'com.testproject.detox'
})
detox[82520] TRACE: [ArtifactsManager.js/LIFECYCLE] artifactsManager.onBeforeLaunchApp({
  bundleId: 'com.testproject.detox',
  deviceId: 'FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C',
  launchArgs: {
    detoxServer: 'ws://localhost:56094',
    detoxSessionId: '3df58e7d-a33e-48b9-fa67-ec0177e11763'
  }
})
detox[82520] DEBUG: [exec.js/EXEC_CMD, #5] SIMCTL_CHILD_DYLD_INSERT_LIBRARIES="/Users/developer/Library/Detox/ios/c148b9f1a94feec75286e47c59476f0083831829/Detox.framework/Detox" /usr/bin/xcrun simctl launch FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C com.testproject.detox --args -detoxServer "ws://localhost:56094" -detoxSessionId "3df58e7d-a33e-48b9-fa67-ec0177e11763" -detoxDisableHierarchyDump "YES"
detox[82520] DEBUG: [exec.js/EXEC_TRY, #5] Launching com.testproject.detox...
detox[82520] TRACE: [exec.js/EXEC_SUCCESS, #5] com.testproject.detox: 82576

detox[82520] DEBUG: [exec.js/EXEC_CMD, #6] /usr/bin/xcrun simctl get_app_container FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C com.testproject.detox
detox[82520] TRACE: [exec.js/EXEC_SUCCESS, #6] /Users/developer/Library/Developer/CoreSimulator/Devices/FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C/data/Containers/Bundle/Application/F03E26E7-1439-4255-9482-B762AD355241/TestProject.app

detox[82520] INFO:  [AppleSimUtils.js] com.testproject.detox launched. To watch simulator logs, run:
        /usr/bin/xcrun simctl spawn FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C log stream --level debug --style compact --predicate 'process == "TestProject"'
detox[82576] TRACE: [ArtifactsManager.js/LIFECYCLE] artifactsManager.onLaunchApp({
  bundleId: 'com.testproject.detox',
  deviceId: 'FC54CC73-AA5B-4C12-A98B-4240A2AD7B5C',
  launchArgs: {
    detoxServer: 'ws://localhost:56094',
    detoxSessionId: '3df58e7d-a33e-48b9-fa67-ec0177e11763',
    detoxDisableHierarchyDump: 'YES'
  },
  pid: 82576
})
detox[82520] TRACE: [AsyncWebSocket.js/WEBSOCKET_SEND] {"type":"isReady","params":{},"messageId":-1000}
detox[82520] TRACE: [DetoxServer.js/MESSAGE] role=tester action=isReady (sessionId=3df58e7d-a33e-48b9-fa67-ec0177e11763)
detox[82520] DEBUG: [DetoxServer.js/CANNOT_FORWARD] role=testee not connected, cannot fw action (sessionId=3df58e7d-a33e-48b9-fa67-ec0177e11763)
detox[82520] DEBUG: [DetoxServer.js/LOGIN] role=testee, sessionId=3df58e7d-a33e-48b9-fa67-ec0177e11763
detox[82520] DEBUG: [DetoxServer.js/LOGIN_SUCCESS] role=testee, sessionId=3df58e7d-a33e-48b9-fa67-ec0177e11763
detox[82520] TRACE: [DetoxServer.js/MESSAGE] role=testee action=AppWillTerminateWithError (sessionId=3df58e7d-a33e-48b9-fa67-ec0177e11763)
detox[82520] TRACE: [AsyncWebSocket.js/WEBSOCKET_MESSAGE] {"type":"AppWillTerminateWithError","messageId":-10000,"params":{"queueName":"com.google.Maps.TileCompletion-TT10","threadNumber":16,"errorDetails":"Signal 11 was raised\n(\n\t0   Detox                               0x00000001122aebf5 __DTXHandleSignal + 59\n\t1   libsystem_platform.dylib            0x00007fff5e7305fd _sigtramp + 29\n\t2   ???                                 0x0000600002774020 0x0 + 105553157636128\n\t3   libc++.1.dylib                      0x00007fff202cd4d9 std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::assign(char const*) + 21\n\t4   TestProject                         0x000000010f6a94a5 gmscore::model::AppendLabelElementsWithLabelElement(GMSx_GMPCLabelElement*, id<GMSStyleTable>, gmscore::model::StyleID const&, gmscore::model::StyleID const, GMPCLegendNamedStyle, gmscore::model::StyleID const&, std::__1::vector<gmscore::model::LabelElement, GMPCLegendNamedStyle::allocator<std::__1::vector> >*) + 283\n\t5   TestProject                         0x000000010f6a91b9 gmscore::model::MakeLabelGroupWithLabelGroup(GMSx_GMPCLabelGroup*, GMSWireContext*, gmscore::model::StyleID const&, GMPCLegendNamedStyle, GMPCLegendNamedStyle) + 542\n\t6   TestProject                         0x000000010f68edf8 +[GMSPointOfInterest unpackedPOIWithProto:wireContext:] + 873\n\t7   TestProject                         0x000000010f6e8cc3 GMSVectorTileFromTileProto + 3562\n\t8   TestProject                         0x000000010f6e7b35 GMSModelTileFromTileProto + 129\n\t9   TestProject                         0x000000010f6f29f5 -[GMSMapTileService processResponseProto:intoBatchResponse:] + 939\n\t10  TestProject                         0x000000010f6ef3ae -[GMSDASHTileService handleServerResponse:error:tileRequests:elapsedTime:] + 268\n\t11  TestProject                         0x000000010f6ead7c __71-[GMSDASHBatchedTileRequest initWithTileRequests:timingClock:delegate:]_block_invoke + 137\n\t12  TestProject                         0x000000010f5294db -[GMSDASHProtobufferRequest scheduleRetryIfApplicableForResponse:error:connection:] + 576\n\t13  TestProject                         0x000000010f528e04 __74-[GMSDASHProtobufferRequest dispatchHandlerWithResponse:error:connection:]_block_invoke + 43\n\t14  libdispatch.dylib                   0x00007fff2010532f _dispatch_call_block_and_release + 12\n\t15  libdispatch.dylib                   0x00007fff20106508 _dispatch_client_callout + 8\n\t16  libdispatch.dylib                   0x00007fff201089fb _dispatch_continuation_pop + 445\n\t17  libdispatch.dylib                   0x00007fff201080bb _dispatch_async_redirect_invoke + 747\n\t18  libdispatch.dylib                   0x00007fff20116074 _dispatch_root_queue_drain + 350\n\t19  libdispatch.dylib                   0x00007fff20116871 _dispatch_worker_thread2 + 102\n\t20  libsystem_pthread.dylib             0x00007fff5e7399f7 _pthread_wqthread + 220\n\t21  libsystem_pthread.dylib             0x00007fff5e738b77 start_wqthread + 15\n)"}}

detox[82520] ERROR: Signal 11 was raised
(
	0   Detox                               0x00000001122aebf5 __DTXHandleSignal + 59
	1   libsystem_platform.dylib            0x00007fff5e7305fd _sigtramp + 29
	2   ???                                 0x0000600002774020 0x0 + 105553157636128
	3   libc++.1.dylib                      0x00007fff202cd4d9 std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::assign(char const*) + 21
	4   TestProject                         0x000000010f6a94a5 gmscore::model::AppendLabelElementsWithLabelElement(GMSx_GMPCLabelElement*, id<GMSStyleTable>, gmscore::model::StyleID const&, gmscore::model::StyleID const, GMPCLegendNamedStyle, gmscore::model::StyleID const&, std::__1::vector<gmscore::model::LabelElement, GMPCLegendNamedStyle::allocator<std::__1::vector> >*) + 283
	5   TestProject                         0x000000010f6a91b9 gmscore::model::MakeLabelGroupWithLabelGroup(GMSx_GMPCLabelGroup*, GMSWireContext*, gmscore::model::StyleID const&, GMPCLegendNamedStyle, GMPCLegendNamedStyle) + 542
	6   TestProject                         0x000000010f68edf8 +[GMSPointOfInterest unpackedPOIWithProto:wireContext:] + 873
	7   TestProject                         0x000000010f6e8cc3 GMSVectorTileFromTileProto + 3562
	8   TestProject                         0x000000010f6e7b35 GMSModelTileFromTileProto + 129
	9   TestProject                         0x000000010f6f29f5 -[GMSMapTileService processResponseProto:intoBatchResponse:] + 939
	10  TestProject                         0x000000010f6ef3ae -[GMSDASHTileService handleServerResponse:error:tileRequests:elapsedTime:] + 268
	11  TestProject                         0x000000010f6ead7c __71-[GMSDASHBatchedTileRequest initWithTileRequests:timingClock:delegate:]_block_invoke + 137
	12  TestProject                         0x000000010f5294db -[GMSDASHProtobufferRequest scheduleRetryIfApplicableForResponse:error:connection:] + 576
	13  TestProject                         0x000000010f528e04 __74-[GMSDASHProtobufferRequest dispatchHandlerWithResponse:error:connection:]_block_invoke + 43
	14  libdispatch.dylib                   0x00007fff2010532f _dispatch_call_block_and_release + 12
	15  libdispatch.dylib                   0x00007fff20106508 _dispatch_client_callout + 8
	16  libdispatch.dylib                   0x00007fff201089fb _dispatch_continuation_pop + 445
	17  libdispatch.dylib                   0x00007fff201080bb _dispatch_async_redirect_invoke + 747
	18  libdispatch.dylib                   0x00007fff20116074 _dispatch_root_queue_drain + 350
	19  libdispatch.dylib                   0x00007fff20116871 _dispatch_worker_thread2 + 102
	20  libsystem_pthread.dylib             0x00007fff5e7399f7 _pthread_wqthread + 220
	21  libsystem_pthread.dylib             0x00007fff5e738b77 start_wqthread + 15
)
detox[82520] INFO:  Example is assigned to undefined
detox[82520] INFO:  Example: should have map view
detox[82520] INFO:  Example: should have map view [SKIPPED]

detox[82520] TRACE: [ArtifactsManager.js/LIFECYCLE] artifactsManager.onBeforeCleanup()
detox[82520] DEBUG: [DetoxServer.js/DISCONNECT] role=tester, sessionId=3df58e7d-a33e-48b9-fa67-ec0177e11763
detox[82520] DEBUG: [DetoxServer.js/DISCONNECT] role=testee, sessionId=3df58e7d-a33e-48b9-fa67-ec0177e11763
detox[82520] DEBUG: [DetoxServer.js/CANNOT_FORWARD] role=tester not connected, cannot fw action (sessionId=3df58e7d-a33e-48b9-fa67-ec0177e11763)
detox[82520] DEBUG: [DetoxServer.js/WS_CLOSE] Detox server connections terminated gracefully
detox[82519] ERROR: [cli.js] Error: Command failed: jest --config e2e/config.json --testNamePattern '^((?!:android:).)*$' --maxWorkers 1 e2e
```

**Comment by bgneu** [comment]
I'm fairly new to Detox as you can tell by my description, so any help regarding this issue will be greatly appreciated.

**Comment by LeoNatan** [comment]
Hello,
I will take a look tomorrow. Please do not delete your test repo.

**Comment by LeoNatan** [comment]
From the crash backtrace, I don't see something related to Detox. Normally, Detox should not be interfering with app code.

**Comment by LeoNatan** [comment]
After setting a correct API key (and changing the bundle ID to match that API key), I only see the following when launching:

![Simulator Screen Shot - iPhone 11 Pro (13 5) - 2020-12-09 at 06 13 43](https://user-images.githubusercontent.com/2270433/101584211-ce844100-39e5-11eb-886c-16a26bc0df64.png)

Detox doesn't crash in this case.

**Comment by LeoNatan** [comment]
Please provide a working example. Thanks

**Comment by bgneu** [comment]
@LeoNatan That's odd... this may sound like a support question but did you make the build after making those updates?

If so, even though it didn't crash did the test pass for you?

**Comment by bgneu** [comment]
FYI I just did all the same steps I shared with you and am still able to consistently repro. I only see what you shared on the screenshot when using an invalid API Key. Could it be that you have a typo on your key, or when you replaced the bundle id?

Meanwhile I'm having other team members run this to check if they are also able to repro on their machines.

**Comment by LeoNatan** [comment]
I have a demo app set in Google. When the bundle identifier didn't match, the GMS SDK complained in the log. After fixing that, there are no complaints, but nothing loads. 

I ran directly from Xcode. So it compiles changes on each run.

**Comment by bgneu** [comment]
If you are running from Xcode (w/ no detox) and nothing loads, I believe there may still a problem with your API key setup because the map should load successfully otherwise. 

Last thing I want to do is have you spend too much time debugging Google Maps setup, so instead I have made the repo private, set everything up and gave you access. So please accept the invite to access the repo. 

You should now be able to do the following:
1. Check out the repo [here](https://github.com/bgneu/DetoxTest).
2. On project root, run `yarn install`.
3. On project root, run `cd ios && pod install && cd ...`
4. On project root, run `detox build -c ios`.
5. On project root, run `detox test -c ios`.
6. Project should crash.

**Comment by LeoNatan** [comment]
Cheers, will try soon. 

Actually, I didn't run the project without Detox linked, but Detox shouldn't interfere with running a test. I'll run first just the project to see that it works, then I'll run with Detox linked, and finally will run a test.

**Comment by LeoNatan** [comment]
Reproduced with your API key. Just linking Detox causes issues. Investigating now.

**Comment by LeoNatan** [comment]
Found the issue, will create a PR with a fix.

**Comment by LeoNatan** [comment]
Fixed in 17.14.4

**Comment by bgneu** [comment]
I'll try the new release soon... thanks for the quick turn around!
